Sources/StreamChat/Database/DTOs/UserDTO.swift

```diff
@@ -23,7 +23,7 @@ class UserDTO: NSManagedObject {
 
     @NSManaged var members: Set<MemberDTO>?
     @NSManaged var currentUser: CurrentUserDTO?
-    @NSManaged var teams: Set<TeamId>?
+    @NSManaged var teams: [TeamId]
```

Comment:
Does this actually work? 🤔 I thought that Core Data only supported NSSet or NSOrderedSet 🤔 
Also, does this mean that we need to change all `Set<MemberDTO>` to `Set<MemberDTO>?` This might also be the reason for the crashes we are having on DTO's. CC @adamrushy

Reply:
Using sets is fine for **relations** but not supported for `Transformable` attributes (at least on `<iOS13` which can be the reason why the crash is not happening does not pop up on later versions). Haven't yet found the docs with regard to `Transformable` on iOS13+ but keep looking

Reply:
Speaking about relations that can cause issues, I believe it's OK with those that have `to many` rule (since the empty set is a valid value) but it is worth checking `to one` for optionality mismatch between attribute and property. 

The same needs to be done for other primitive attributes. When I was debugging this crash I found some inconsistencies (e.g. `UserDTO.isOnline` is non-optional but the default value is `none`)
<img width="1230" alt="Screenshot 2022-02-09 at 20 00 56" src="https://user-images.githubusercontent.com/12818985/153251572-0ba40a56-c6fc-4343-b527-e1830fadea00.png">

Reply:
@evsaev Nice find! If you create a ticket please link it to this one: CIS-1558